dotCMS/src/main/java/com/dotmarketing/util/ZipUtil.java

```diff
@@ -82,13 +81,16 @@ public static void extract(final ZipFile zipFile, final File toDir) throws IOExc
 		   if (!toDir.exists()){
 		   		toDir.mkdirs();
 		   }
-		   final Enumeration entries = zipFile.entries();
+		   final Enumeration<?> entries = zipFile.entries();
 		   while (entries.hasMoreElements()) {
 			   final ZipEntry zipEntry = (ZipEntry) entries.nextElement();
 			   if (zipEntry.isDirectory()) {
 				   final File dir = new File(toDir, zipEntry.getName());
-				   if (checkSecurity(toDir,dir) && !dir.exists()){ // make sure also empty directories get created!
+				   if (dir.getCanonicalPath().startsWith(toDir.getCanonicalPath()) && !dir.exists()){ // make sure also empty directories get created!
```

Comment:
I think this code will conflict with the changes coming in 22.12 and this issue: https://github.com/dotCMS/core/issues/23401

https://github.com/dotCMS/core/blob/release-22.12/dotCMS/src/main/java/com/dotmarketing/util/ZipUtil.java

Reply:
I saw your fixes on that and this has already been merged with it. In fact, I had based my fixes on the work you did on that branch. Somehow SQ said it was assigned to you @wezell . And I got to your branch from there. 
I know a simple call to checkSecurity takes care of the issue. But SQ does not pick it up. It keeps saying it isn't complying.  The new code still uses securityCheck once an issue is found.

Reply:
I guess what is important to me is that we have 1 code path for unzipping stuff rather than a few.  Is this still the case with your changes?  If so, all good.

Reply:
We're good. I just double-checked.